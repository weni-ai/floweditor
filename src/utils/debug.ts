import { FlowStoreProps } from 'components/flow/Flow';
import { DebugState } from 'store/editor';

import mutate from 'immutability-helper';

/* istanbul ignore next -- @preserve */
export default class Debug {
  private props: FlowStoreProps;
  private state: DebugState;

  constructor(props: FlowStoreProps, initial: DebugState) {
    this.props = props;
    this.state = initial || { showUUIDs: false };
  }

  public showUUIDs(): DebugState {
    const updated = mutate(this.state, { $merge: { showUUIDs: true } });
    this.props.mergeEditorState({ debug: updated });
    return updated;
  }
}
