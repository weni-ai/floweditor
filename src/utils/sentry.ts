import * as Sentry from '@sentry/react';

const initSentry = (dsn: string) => {
  Sentry.init({
    dsn,
    integrations: [
      Sentry.browserTracingIntegration(),
      Sentry.httpClientIntegration({
        failedRequestStatusCodes: [[400, 499], [500, 599]],
      }),
      Sentry.replayIntegration(),
      Sentry.replayCanvasIntegration(),
    ],
    tracesSampleRate: 1.0,
    replaysSessionSampleRate: 0.1,
    replaysOnErrorSampleRate: 1.0,
  });
};

export default initSentry;
