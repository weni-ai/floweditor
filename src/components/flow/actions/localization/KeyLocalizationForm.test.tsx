import * as React from 'react';
import { fireEvent, render, fireUnnnicTextAreaChangeText } from 'test/utils';
import {
  createSendEmailAction,
  getLocalizationFormProps,
  Spanish,
} from 'testUtils/assetCreators';

import KeyLocalizationForm from './KeyLocalizationForm';

describe(KeyLocalizationForm.name, () => {
  it('renders send email', () => {
    const props = getLocalizationFormProps(createSendEmailAction());
    const { baseElement } = render(<KeyLocalizationForm {...props} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('handles changes', () => {
    const props = getLocalizationFormProps(createSendEmailAction());
    const { getByTestId, getByText } = render(
      <KeyLocalizationForm {...props} />,
    );

    // modify the subject
    fireUnnnicTextAreaChangeText(getByTestId('Subject'), 'translated subject');

    // modify the body
    fireEvent.click(getByText('Body Translation'));
    fireUnnnicTextAreaChangeText(getByTestId('Body'), 'translated body');

    // save our translations
    fireEvent.click(getByText('Ok'));
    expect(props.updateLocalizations).toMatchSnapshot();
  });

  it('removes translations', () => {
    const action = createSendEmailAction();
    const props = getLocalizationFormProps(action, Spanish, {
      [action.uuid]: { subject: ['hola'] },
    });

    // show that we initialized with hola
    const { baseElement, getByTestId, getByText } = render(
      <KeyLocalizationForm {...props} />,
    );
    expect(baseElement).toMatchSnapshot();

    // clear the translation
    fireUnnnicTextAreaChangeText(getByTestId('Subject'), '');

    // save our translations, which should remove the key
    fireEvent.click(getByText('Ok'));
    expect(props.updateLocalizations).toMatchSnapshot();
  });

  it('should close the modal', () => {
    const props = getLocalizationFormProps(createSendEmailAction());
    const { getByText } = render(<KeyLocalizationForm {...props} />);

    fireEvent.click(getByText('Cancel'));
    expect(props.onClose).toHaveBeenCalled();
  });
});
