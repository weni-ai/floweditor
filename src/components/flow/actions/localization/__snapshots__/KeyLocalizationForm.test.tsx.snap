// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KeyLocalizationForm > handles changes 1`] = `
[MockFunction spy] {
  "calls": [
    [
      "eng",
      [
        {
          "translations": {
            "body": "translated body",
            "subject": "translated subject",
          },
          "uuid": "763e4844-3e1b-407a-a1b5-5fdfcd308b41",
        },
      ],
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`KeyLocalizationForm > removes translations 1`] = `
<body>
  <div>
    <div
      class="_dialog_af1b5d"
    >
      <div
        class="_header_af1b5d"
      >
        <div
          class="_title_container_af1b5d"
        >
          <div
            class="_title_af1b5d u font secondary title-sm black"
          >
            Send email
          </div>
          <div
            class="_subtitle_af1b5d"
          />
        </div>
        <div
          data-use-vue-component-wrap=""
          data-v-app=""
          style="all: unset;"
        >
          <div
            class="tab size-sm _tabs_af1b5d"
            data-v-54432ea7=""
          >
            <header
              class="tab-header"
              data-v-54432ea7=""
            >
              <ul
                class="tab-content"
                data-v-54432ea7=""
              >
                
                <li
                  class="tab-head tab-head--active"
                  data-v-54432ea7=""
                >
                  
                  <div
                    __use_react_component_wrap=""
                    style="all: unset;"
                  >
                    General
                  </div>
                  
                </li>
                <li
                  class="tab-head"
                  data-v-54432ea7=""
                >
                  
                  <div
                    __use_react_component_wrap=""
                    style="all: unset;"
                  >
                    Body Translation
                  </div>
                  
                </li>
                
              </ul>
            </header>
            <main
              class="tab-body"
              data-v-54432ea7=""
            >
              <div
                class="tab-panel"
                data-v-54432ea7=""
              >
                
                
              </div>
            </main>
          </div>
        </div>
      </div>
      <div
        class="_content_af1b5d"
      >
        <div>
          <div
            data-spec="translation-container"
          >
            <div
              class="_translate_from_604508"
              data-spec="text-to-translate"
            >
              New Sign Up
            </div>
          </div>
          <div>
            <div
              class="_completion_wrapper_17662c"
            >
              <div
                data-use-vue-component-wrap=""
                data-v-app=""
                style="all: unset;"
              >
                <div
                  class="unnnic-text-area sm normal _textarea_17662c"
                  data-testid="Subject"
                  data-v-e1948bc4=""
                >
                  <!---->
                  <textarea
                    data-v-e1948bc4=""
                    placeholder="Spanish Translation"
                  />
                  <!---->
                  <!---->
                </div>
              </div>
              <div
                class="_options_wrapper_08dd12"
                data-testid="temba_completion_options_Subject"
                id=""
                style="margin-top: 1px; width: -24px; display: none;"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="_footer_af1b5d"
      >
        <div
          class="_buttons_af1b5d"
        >
          <div
            class="_buttons_wrapper_af1b5d"
          >
            <div
              class="_right_buttons_af1b5d"
            >
              <div
                data-use-vue-component-wrap=""
                data-v-app=""
              >
                <button
                  class="unnnic-button unnnic-button--size-large unnnic-button--tertiary"
                  data-v-d3a33348=""
                  style="margin-left: 0px; margin-top: 0px; margin-right: 0px;"
                >
                  <!---->
                  <!---->
                  <!---->
                  <span
                    class="unnnic-button__label"
                    data-v-d3a33348=""
                  >
                    
                    
                     Cancel
                  </span>
                  <!---->
                </button>
              </div>
              <div
                data-use-vue-component-wrap=""
                data-v-app=""
              >
                <button
                  class="unnnic-button unnnic-button--size-large unnnic-button--primary"
                  data-v-d3a33348=""
                  style="margin-top: 0px; margin-right: 0px;"
                >
                  <!---->
                  <!---->
                  <!---->
                  <span
                    class="unnnic-button__label"
                    data-v-d3a33348=""
                  >
                    
                    
                     Ok
                  </span>
                  <!---->
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`KeyLocalizationForm > removes translations 2`] = `
[MockFunction spy] {
  "calls": [
    [
      "spa",
      [
        {
          "translations": {},
          "uuid": "132de855-4042-4dc1-a18f-cc2e6a8f790a",
        },
      ],
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`KeyLocalizationForm > renders send email 1`] = `
<body>
  <div>
    <div
      class="_dialog_af1b5d"
    >
      <div
        class="_header_af1b5d"
      >
        <div
          class="_title_container_af1b5d"
        >
          <div
            class="_title_af1b5d u font secondary title-sm black"
          >
            Send email
          </div>
          <div
            class="_subtitle_af1b5d"
          />
        </div>
        <div
          data-use-vue-component-wrap=""
          data-v-app=""
          style="all: unset;"
        >
          <div
            class="tab size-sm _tabs_af1b5d"
            data-v-54432ea7=""
          >
            <header
              class="tab-header"
              data-v-54432ea7=""
            >
              <ul
                class="tab-content"
                data-v-54432ea7=""
              >
                
                <li
                  class="tab-head tab-head--active"
                  data-v-54432ea7=""
                >
                  
                  <div
                    __use_react_component_wrap=""
                    style="all: unset;"
                  >
                    General
                  </div>
                  
                </li>
                <li
                  class="tab-head"
                  data-v-54432ea7=""
                >
                  
                  <div
                    __use_react_component_wrap=""
                    style="all: unset;"
                  >
                    Body Translation
                  </div>
                  
                </li>
                
              </ul>
            </header>
            <main
              class="tab-body"
              data-v-54432ea7=""
            >
              <div
                class="tab-panel"
                data-v-54432ea7=""
              >
                
                
              </div>
            </main>
          </div>
        </div>
      </div>
      <div
        class="_content_af1b5d"
      >
        <div>
          <div
            data-spec="translation-container"
          >
            <div
              class="_translate_from_604508"
              data-spec="text-to-translate"
            >
              New Sign Up
            </div>
          </div>
          <div>
            <div
              class="_completion_wrapper_17662c"
            >
              <div
                data-use-vue-component-wrap=""
                data-v-app=""
                style="all: unset;"
              >
                <div
                  class="unnnic-text-area sm normal _textarea_17662c"
                  data-testid="Subject"
                  data-v-e1948bc4=""
                >
                  <!---->
                  <textarea
                    data-v-e1948bc4=""
                    placeholder="English Translation"
                  />
                  <!---->
                  <!---->
                </div>
              </div>
              <div
                class="_options_wrapper_08dd12"
                data-testid="temba_completion_options_Subject"
                id=""
                style="margin-top: 1px; width: -24px; display: none;"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="_footer_af1b5d"
      >
        <div
          class="_buttons_af1b5d"
        >
          <div
            class="_buttons_wrapper_af1b5d"
          >
            <div
              class="_right_buttons_af1b5d"
            >
              <div
                data-use-vue-component-wrap=""
                data-v-app=""
              >
                <button
                  class="unnnic-button unnnic-button--size-large unnnic-button--tertiary"
                  data-v-d3a33348=""
                  style="margin-left: 0px; margin-top: 0px; margin-right: 0px;"
                >
                  <!---->
                  <!---->
                  <!---->
                  <span
                    class="unnnic-button__label"
                    data-v-d3a33348=""
                  >
                    
                    
                     Cancel
                  </span>
                  <!---->
                </button>
              </div>
              <div
                data-use-vue-component-wrap=""
                data-v-app=""
              >
                <button
                  class="unnnic-button unnnic-button--size-large unnnic-button--primary"
                  data-v-d3a33348=""
                  style="margin-top: 0px; margin-right: 0px;"
                >
                  <!---->
                  <!---->
                  <!---->
                  <span
                    class="unnnic-button__label"
                    data-v-d3a33348=""
                  >
                    
                    
                     Ok
                  </span>
                  <!---->
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
